### PR TITLE
adding btagEff maps for TtH signal samples

### PR DESCRIPTION
**Pull request by EJDomi**

### CODE REVIEW

**Review by dmajumder — CHANGES_REQUESTED** [review]
@EJDomi 
The ttjets efficiency is probably going to be different from the signal. 

But for the signal, are the efficiencies for the different mass points appreciably different? If not, all signal efficiency maps can be combined into one.

Did you make the eff maps for the other non-QCD bkgs?

**Comment by EJDomi** [comment]
@dmajumder 
The efficiencies are similar, but the statistical error pulled from the histograms are not within range of each other, does this matter? If not, then I can just make a combined signal map
I have not made the eff-maps for the other backgrounds, but they can be quickly made

@EJDomi I suggest having one eff map for all the signals. As for teh backgrounds, let's use the ttjets eff map for now. The other backgrounds are small.
For the control region, we may need a Wjets eff map.

File: python/JetMaker_cfi.py [hunk 1]
```diff
@@ -90,7 +90,7 @@
 
 ### https://twiki.cern.ch/twiki/bin/view/CMS/JetWtagging#Recommendation_for_13_TeV_data_a (W tag)
 defaultCHSHJetSelectionParameters = defaultAK8CHSJetSelectionParameters.clone(
-    jetPtMin            = cms.double(300),
+    jetPtMin            = cms.double(200),
     jetPrunedMassMin    = cms.double(105) ,
```
Comment (dmajumder) [inline python/JetMaker_cfi.py]: The Higgs pT should be > 300 GeV. Please put it back.
The Z pT can be > 200 GeV

File: python/VLQAna_cfi.py [hunk 2]
```diff
@@ -45,6 +45,16 @@
       subjetHighestCSVMin = cms.double(-1000000),
       subjetHighestCSVMax = defaultCHSTJetSelectionParameters.subjetHighestCSVMin, 
       ),
+    jetZTaggedselParams        = defaultCHSHJetSelectionParameters.clone(
+      jetPrunedMassMin = cms.double(65.),
```
Comment (dmajumder) [inline python/VLQAna_cfi.py]: Here update the Z pT cut to 200 GeV

File: python/VLQAna_cfi.py [hunk 3]
```diff
@@ -45,6 +45,16 @@
...
+      jetPrunedMassMax = cms.double(105.),
+      ),
+    jetAntiZTaggedselParams        = defaultCHSHJetSelectionParameters.clone(
+      jetPrunedMassMin = cms.double(65.),
```
Comment (dmajumder) [inline python/VLQAna_cfi.py]: Here update the Z pT cut to 200

File: src/JetMaker.cc [hunk 4]
```diff
@@ -521,7 +521,9 @@ void JetMaker::operator()(edm::Event& evt, vlq::JetCollection& jets) {
             jet.setNConsts      ( (h_jetcMultip.product())->at(ijet) + (h_jetnMultip.product())->at(ijet) ) ; 
             jet.setGroomedMassCorr (masssmear * massCorr * (1 + jecShift_*unc) ) ; 
             jet.setNSubjets (nsubjets) ; 
-            jet.setNSubjetsBTaggedCSVL (nsubjetsbtaggedcsvl) ; 
+            jet.setNSubjetsBTaggedCSVL (nsubjetsbtaggedcsvl) ;
+            if (ijet == 0) jet.setIsleading(true);
+            else if (ijet == 1) jet.setIs2ndleading(true) ;
```
Comment (dmajumder) [inline src/JetMaker.cc:526]: I need to understand why this is needed

File: plugins/VLQAna.cc [hunk 5]
```diff
@@ -289,63 +295,102 @@ bool VLQAna::filter(edm::Event& evt, const edm::EventSetup& iSetup) {
...
-    theTop = std::unique_ptr<vlq::Jet>(new vlq::Jet(goodTopTaggedJets.at(0))) ; 
 
-    if (nHiggs > 0) {
-      for (vlq::Jet& hjet : goodHTaggedJets) {
-        double dphi = abs((theTop->getP4()).DeltaPhi(hjet.getP4())) ;  
-        if (dphi > 2.0) {
-          theHiggs = std::unique_ptr<vlq::Jet>(new vlq::Jet(hjet)) ; 
-          isRegionD = true ; 
-          break ; 
+  if (nTop > 0) {
```
Comment (dmajumder) [inline plugins/VLQAna.cc:329]: This manipulation for the ABCD region is best done in the anaTp.py
In that case, you just store the collections for top, anti-top, Higgs, anti-Higgs, Z, anti-Z jets in the tree.

**Review by dmajumder — CHANGES_REQUESTED** [review]

File: src/JetMaker.cc (hunk 4 above)
Comment (EJDomi) [reply src/JetMaker.cc:526]: Its purpose was to be able to check in VLQAna.cc, whether a jet in one of the vectors of 'goodTaggedJets' such as the 'goodTopTaggedJets' was the leading AK8 jet. That was the easiest way in which I could think of doing it, though there may be problems in how it is implemented

**Review by dmajumder — APPROVED** [review]

**Pull request merged**